src/core/datagram.c

```diff
@@ -0,0 +1,434 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    Manages the unreliable datagram feature's functionality for a connection.
+
+--*/
+
+#include "precomp.h"
+
+#ifdef QUIC_LOGS_WPP
+#include "datagram.tmh"
+#endif
+
+#define DATAGRAM_FRAME_HEADER_LENGTH 3
+
+#define QUIC_DATAGRAM_OVERHEAD(CidLength) \
+(\
+    MIN_SHORT_HEADER_LENGTH_V1 + \
+    CidLength + \
+    DATAGRAM_FRAME_HEADER_LENGTH \
+)
+
+uint16_t
+QuicCalculateDatagramLength(
+    _In_ QUIC_ADDRESS_FAMILY Family,
+    _In_ uint16_t Mtu,
+    _In_ uint8_t CidLength
+    )
+{
+    return
+        MaxUdpPayloadSizeForFamily(Family, Mtu) -

```

Comment:
Have you checked this returns the expected value? You might want to sum the smaller values, and wrap them in parens and only have one subtraction operation.

Reply:
What do you think wouldn't work here?

Reply:
I recently had an issue where I had `A - B - C`, but the result was `A - (B - C)`, instead of the desired result of `(A - B) - C`.  Perhaps there was a subtle difference in my situation, that triggered the unexpected result. Just asking to confirm here that the result is the desired one.

Reply:
It all seems to work as expected.